### PR TITLE
IBC Governance Action Support

### LINKED ISSUE #2173: IBC-related governance proposals
We don't want to forget to have governance proposals for modifying / mutating IBC state:

- [ ] Performing unplanned upgrades of an IBC connection
- [ ] Freezing IBC clients
- [ ] Unfreezing IBC clients

### PR DESCRIPTION
**Pull request by zbuc**
Closes #2173

### CODE REVIEW

File: crates/core/app/src/action_handler/actions/submit.rs [hunk 1]
```diff
@@ -203,6 +223,30 @@ impl ActionHandler for ProposalSubmit {
             ProposalPayload::UpgradePlan { .. } => {
                 // TODO(erwan): no stateful checks for upgrade plan.
             }
+            ProposalPayload::UnplannedIbcUpgrade { .. } => {
+                // TODO: which checks happen here?
+                todo!();
```
Comment (erwanor) [inline crates/core/app/src/action_handler/actions/submit.rs:228]: I suggest we ship this PR with Freeze/Unfreeze and leave out trying to handle arbitrary client modifications for now, and remove the `todo!()`. The reason for this is that we do not have a great idea yet of the both the scope of changes that need to happen there.

Comment (avahowell) [reply crates/core/app/src/action_handler/actions/submit.rs:228]: Agree with this

**Review by avahowell — CHANGES_REQUESTED** [review]

**Comment by conorsch** [comment]
We are letting this slip from Testnet 64, and will return after shipping.

File: crates/core/component/governance/src/proposal.rs [hunk 2]
```diff
@@ -202,6 +212,15 @@ pub enum ProposalKind {
     /// An upgrade proposal.
     #[cfg_attr(feature = "clap", clap(display_order = 500))]
     UpgradePlan,
+    /// An unplanned IBC connection modification.
+    #[cfg_attr(feature = "clap", clap(display_order = 600))]
+    UnplannedIbcUpgrade,
```
Comment (avahowell) [inline crates/core/component/governance/src/proposal.rs:217]: should this be removed now?

File: crates/core/component/governance/src/component/view.rs [hunk 3]
```diff
@@ -875,8 +879,30 @@ pub trait StateWriteExt: StateWrite {
...
+                let client_id = &ClientId::from_str(client_id)
+                    .map_err(|e| tonic::Status::aborted(format!("invalid client id: {e}")))?;
+                let current_height = self
+                    .get_block_height()
+                    .await
+                    .expect("block height should be set");
+                let client_state = self.get_client_state(client_id).await?;
+                let revision_number = self.get_revision_number().await?;
+
+                let freeze_height = IbcHeight::new(revision_number, current_height)?;
```
Comment (avahowell) [inline crates/core/component/governance/src/component/view.rs:892]: I think this height is wrong, it should be the height at which the client was frozen, in terms of that client's height (not the host chain's height). So we should be pulling the height from the client_state here instead.

**Review by avahowell — CHANGES_REQUESTED** [review]

File: crates/core/component/governance/src/component/view.rs (hunk 3 above)
Comment (erwanor) [reply crates/core/component/governance/src/component/view.rs]: Ah yes, makes sense

**Review by avahowell — APPROVED** [review]

**Pull request merged**